src/core/border_router/routing_manager.cpp

```diff
@@ -953,7 +953,8 @@ bool RoutingManager::IsValidBrUlaPrefix(const Ip6::Prefix &aBrUlaPrefix)
 
 bool RoutingManager::IsValidOmrPrefix(const NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig)
 {
-    return IsValidOmrPrefix(aOnMeshPrefixConfig.GetPrefix()) && aOnMeshPrefixConfig.mSlaac && !aOnMeshPrefixConfig.mDp;
+    return IsValidOmrPrefix(aOnMeshPrefixConfig.GetPrefix()) && aOnMeshPrefixConfig.mOnMesh &&
+           aOnMeshPrefixConfig.mSlaac && !aOnMeshPrefixConfig.mDp;
```

Comment:
Not directly related to this PR, but one idea/thought on OMR prefix in general.

Should we require OMR prefix to be marked as stable? The reason is if we have an ED which only gets the stable netdata subset, it may not learn of an OMR prefix which is not marked as stable and not get an IPv6 address (somewhat defeating the main purpose behind OMR prefix). 

Thoughts?

Reply:
I think we should require it to be stable.

Reply:
It's already defined in Spec 9.6.1.1 that P_stable should be true. 

So, I think we can add such check in this PR too.